gomail.go

```diff
@@ -100,6 +100,8 @@ const (
 	// Unencoded can be used to avoid encoding the body of an email. The headers
 	// will still be encoded using quoted-printable encoding.
 	Unencoded Encoding = "8bit"
+	// Base64Encoded represents data that has already been base64 encoded
+	Base64Encoded Encoding = "base64encoded"
```

Comment:
Will this string ever be used as output in the mail message? 

Also the naming is a bit confusing.  E.g. Base64 Encoding and Base64Encoded Encoding sound the same.  Maybe call it Base64PreEncoded and Base64PostEncoded or something to differentiate.  Also please update the comment on Base64 to indicate that the base64 encoding will be performed by gomail if that Encoding is used.